networkml/algorithms/host_footprint.py

```diff
@@ -110,6 +137,42 @@ def parse_args(raw_args=None):
         parsed_args = parser.parse_args(raw_args)
         return parsed_args
 
+    def _get_test_train_csv(self, path):
+        df = pd.read_csv(path)
+        df = self.reorder_drop_cols(df)
+        df = df.fillna(0)
+        # Split dataframe into X (the input features or predictors)
+        # and y (the target or outcome or dependent variable)
+        df['role'] = df.filename.str.split('-').str[0]
+        # Drop unknown roles.
+        df = df[df['role'] != 'Unknown']
+        X = df.drop(['filename', 'role'], axis=1)
+        y = df.role
+        X = self.string_feature_check(X)
+        return (X, y)
+
+    def compare_test_data(self, model, scaler, label_encoder, test_data):
+        X_test, y_test = self._get_test_train_csv(test_data)
```

Comment:
This looks good to me. I have a small suggestion. It would be more in line with sklearn documentation and perhaps more readable to persons accustomed to data science terminology to rename ```y_test``` to ```y_true```. Nice job, @anarkiwi

Reply:
Done